### ISSUE TITLE
`Create New Folder` and `Empty Document` do not respect umask settings.

### ISSUE THREAD

**Issue by gusbrs** [linuxmint/nemo#1909]
```
 * Nemo version (nemo --version)
nemo 3.8.5
 * Is issue with desktop or windowed nemo?
desktop
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Mint 19
 * Graphics hardware *and* driver used
(Not sure, but I don't see how it is related. But I can provide it if needed, with some guidance).
 * 32 or 64 bit
64 bit
 ```

**Issue**
`Create New Folder` and `Create New Document` -> `Empty Document` do not respect umask settings.

**Steps to reproduce**

```
~/test$ umask
0077
~/test$ touch test1
~/test$ mkdir testfolder
```
Then, go to Nemo, right click `Create New Folder` and `Create New Document` -> `Empty Document`
```
~/test$ ls -la
total 16
drwx------  4 gustavo gustavo 4096 Jul 24 19:45  .
drwxr-xr-x 39 gustavo gustavo 4096 Jul 24 19:43  ..
-rw-------  1 gustavo gustavo    0 Jul 24 19:44  test1
drwx------  2 gustavo gustavo 4096 Jul 24 19:45  testfolder
-rw-rw-r--  1 gustavo gustavo    0 Jul 24 19:44 'Untitled Document'
drwxrwxr-x  2 gustavo gustavo 4096 Jul 24 19:44 'Untitled Folder'
```

**Expected behaviour**

That the current user umask be respected by nemo in creating new folders and new empty documents.

**Other information**

**Comment by mtwebster** [comment]
Just tested this, nemo apparently needs to be restarted before it will respect a new umask.

```
drwxrwxr-x   2 mtwebster mtwebster      4096 Jul 24 19:08  umask2
drwx------   2 mtwebster mtwebster      4096 Jul 24 19:08  umask3
-rw-------   1 mtwebster mtwebster         0 Jul 24 19:09  umask4
```
The last two I made *after* restarting nemo.

Can you confirm this?

**Comment by gusbrs** [comment]
@mtwebster Thank you for your prompt answer. No, I cannot reproduce the behaviour you describe. As a matter of fact, my umask is set at login (currently in ~/.bashrc) so before nemo is launched in the first place. For the reported testing above, I didn't play with the umask already set.

**Comment by mtwebster** [comment]
Ah, add it to ~/.profile as well and nemo will pick it up

.bashrc will only apply it in shell sessions (terminal).

You'll have to log out and back in.

**Comment by gusbrs** [comment]
Mmh, indeed! I guess I have to get a better grasp on the roles of the config files... And now I know why I only stumbled in this now. I used to have the umask in ~/.profile and things worked as expected. But, in installing Mint 19, I got a umask 0022 in the terminal even when setting 0077 in ~/.profile, so I moved it to ~/.bashrc. I didn't suppose I'd have to set it in both of them. Anyway, thank you very much, and sorry for the undue fuss.

**Comment by mtwebster** [comment]
Actually .profile *should* work for everything - I think that's maybe a real issue here in gnome-terminal (the default in mint, assuming that's what you're using.)

I tried another terminal program and it shows the umask set in .profile

**Comment by mtwebster** [comment]
Looks like this is the bug:
https://bugs.launchpad.net/ubuntu/+source/gnome-terminal/+bug/1685754

Of the solutions offered, it seems simply adding to .bashrc as well as .profile for now is the simplest solution.

Please leave this bug open for now.

**Comment by gusbrs** [comment]
You are right once more. Indeed, with the umask set in ~/.profile, things work as expected for nemo but I find umask 0022 in gnome-terminal (while it is correct in, e.g. Xfce Terminal). Thank you very much again!

**Comment by mtwebster** [comment]
Just be aware that nemo can be launched via dbus (which seems like the common denominator in that bug report) - this most often happens when it's launched by something like Firefox (open location when downloading a file) or other apps that have some sort of 'open file location' menu item.  When that happens nemo will *not* have the umask set in .profile.

This will only happen if nemo isn't already running in another window.  If it is already running, the dbus launch just uses the existing process (which has the correct umask)

**Comment by gusbrs** [comment]
I'll pay attention to that. Thanks yet again.

**Comment by CottonEaster** [comment]
Sorry for my ignorance, but is this - or might this - be why the 'create new <whatever>' option is greyed-out on network shares to which I have write access?

**Comment by xenopeek** [comment]
@mtwebster I think this can be closed? I retested both cases on Linux Mint 20.

Manually starting Nemo:

1. I edited `~/.profile` and added `umask 037`;
2. Logged out & back in;
3. Immediately started Nemo and I created a new file;
4. This correctly gets `-rw-r-----` permission.
5. (This can also be tested from the terminal by running `umask 037 && nemo -q && nemo` then creating a new file in Nemo.)

Indirectly starting Nemo from another program:

1. Re-edited `~/.profile` and set `umask 077` as further test;
2. Logged out & back in;
3. Immediately started Firefox and selected Shows Downloads Folder;
4. Nemo opens, I created a new file here;
5. This correctly gets `-rw-------` permissions.

Both tests look like Nemo correctly gets umask from profile. I also tried the both after disabling nemo-desktop (`gsettings set org.nemo.desktop show-desktop-icons 'false'`), as at log in no processes had already been started. All looks good to me.

As noted in earlier discussion umask on in the shell can differ from the one set in `~/.profile`, that's shell initialization complexity and unrelated to Nemo.

**Comment by gusbrs** [comment]
Hi @xenopeek , thanks for keeping track of this issue.

If considered from the side of Nemo exclusively, you are perfectly correct. The issue initially arose as *gnome-terminal* did not catch the umask set in `~/.profile`, and this led me to (incorrectly) set it in `~/.bashrc` instead, and for this reason Nemo did not get the umask I expected. But already setting it in `~/.profile` in Mint 19 works for Nemo.

That said, and as mtwebster made me understand, the actual problem was (is?) that `gnome-terminal` was not honouring the umask setting in `~/.profile`. And as far as I get it, this was the reason this was kept open since then.

If your point is that this is not an issue with Nemo and should be closed here and reported elsewhere (it already is, as linked by mtwebster), I'm fine with that, of course. But I wouldn't say that the "umask in the shell can differ from the one set in `~/.profile`", as it would be expected that it didn't, as far as I understand.

**Comment by xenopeek** [comment]
Ah. I see, yes Gnome-Terminal (and probably all VTE based terminals; I saw the same on Tilix) ignores umask set in the `~/.profile` file. Other terminals I tried (Terminology and Xfce4-Terminal) correctly use umask set in the `~/.profile` file.

As per my understanding (from the [manual](https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html#Bash-Startup-Files)) when you log in to Linux Mint a login shell initialization is done, reading `/etc/profile` and afterwards the first that exists of `~/.bash_profile`, `~/.bash_login` and `~/.profile`. The umask set last in that is in effect for programs you start in your desktop session. But programs can set their own umask.

When you open a terminal the interactive non-login shell initialization is done, reading `~/.bashrc`. Any umask set in is only in effect for that shell session and programs started from it. If no umask is set you would expect that the umask that is in effect for your desktop session—as per above loaded from `~/.profile`—is set also for your shell session. But Gnome-Terminal sets its own umask in this case.

Long way round to say Nemo doesn't set umask itself: it inherits it from the process that started it. As shown if you start Nemo manually though a menu launcher it correctly inherits umask from your desktop session (originally from `~/.profile`) and the same when starting it indirectly through for example Firefox, which also inherited umask from your desktop session. If you would start Nemo manually by running `nemo` in Gnome-Terminal the umask set for your desktop session isn't applied and if this is the first Nemo processes being run it inherits umask set either from `~/.bashrc` or if such isn't set the default umask that Gnome-Terminal set itself.

As I read it this isn't a issue Nemo should fix.

But it's good to note that if you use a VTE based terminal like Gnome-Terminal, and you've set umask for your desktop session, you should apparently set the same in `~/.bashrc` to apply it also to your shell session for programs you run on or from the terminal. Why there's a difference between terminals in this regard IDK.

**Comment by gusbrs** [comment]
I agree with your analysis.  Whether this issue should remain open because of the behaviour of Gnome-Terminal, I defer to you guys, and I'll be happy with what you decide.

PS: Just saw the new release, I hope to be able to install it soon. Thank you very much to all the team!
